### PR TITLE
PCH Excerpt Generator: Update UI

### PR DESCRIPTION
**Pull request by acicovic**
## Description

This PR refreshes the UI of our PCH Excerpt Generator feature. It also implements a loading animation as described [here](https://developer.wordpress.org/block-editor/reference-guides/components/animate/#loading).

### Deviations from design proposal
- The `WRITE AN EXCERPT (OPTIONAL)` textarea label was left as-is, since this value matches the WordPress default (we can change it if needed). Because the given text this is a label, it cannot have two colors and custom spacing without weird workarounds.
- An external icon to the `Learn more about manual excerpts` link has been added, since it opens in external mode (we can change it if needed).

## Motivation and context
- Update the Excerpt Generator according to the UI proposal.
- To be merged into #2238.

## How has this been tested?
Existing tests pass.

### Current UI
| Empty | Suggestion | Loading |
|--------|--------|--------|
| image | image | image |

### Updated UI
| Empty | Suggestion | Loading |
|--------|--------|--------|
| image | image | image | 

Note: The button will display `Generate Excerpt` when in its initial state. If an excerpt is generated, the button will display `Regenerate Excerpt`.

### CODE REVIEW

File: src/content-helper/excerpt-generator/excerpt-generator.scss [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
-		gap: to_rem(15px);
+		gap: to_rem(8px);
+
+		button {
+			flex-grow: 1;
+			height: to_rem(40px);
+			justify-content: center;
+			margin-bottom: to_rem(8px);
+			margin-top: to_rem(8px);
+			padding: var(--grid-unit-10, 8px) var(--grid-unit-15, 12px);
```
Comment (vaurdan) [inline src/content-helper/excerpt-generator/excerpt-generator.scss]: In the other PRs I have been converting the fallback values to rem with `to_rem` as well. Not sure if it's strictly necessary, but that way we can keep consistency with using rem values even when the variable is undefined.

What do you think? 
```suggestion
			padding: var(--grid-unit-10, to_rem(8px)) var(--grid-unit-15, to_rem(12px));
```

**Review by vaurdan — COMMENTED** [review]
Looks pretty neat! Thank you!

I added a few minor comments, feel free to take a look.

File: src/content-helper/excerpt-generator/excerpt-generator.scss [hunk 2]
```diff
@@ -1,47 +1,55 @@
...
-	background: rgba(255, 255, 255, 0.7); // semi-transparent white background.
-	z-index: 10;
+	top: to_rem(29px);
+	left: to_rem(9px);
 }
 
-.editor-post-excerpt__spinner.has-word-count {
-	bottom: to_rem(20px);
+.editor-post-excerpt__textarea {
+	margin-bottom: to_rem(8px);
```
Comment (vaurdan) [inline src/content-helper/excerpt-generator/excerpt-generator.scss]: I would change this to `var(--grid-unit-10, to_rem(8px));`

```suggestion
	margin-bottom: var(--grid-unit-10, to_rem(8px));
```

File: src/content-helper/excerpt-generator/excerpt-generator.scss [hunk 3]
```diff
@@ -1,47 +1,55 @@
...
-		justify-content: center;
-		align-items: center;
-		gap: to_rem(15px);
+		gap: to_rem(8px);
+
+		button {
+			flex-grow: 1;
+			height: to_rem(40px);
+			justify-content: center;
+			margin-bottom: to_rem(8px);
```
Comment (vaurdan) [inline src/content-helper/excerpt-generator/excerpt-generator.scss]: Same as above.

```suggestion
			margin-bottom: var(--grid-unit-10, to_rem(8px));
			margin-top: var(--grid-unit-10, to_rem(8px));
```

**Comment by acicovic** [comment]
@vaurdan, thanks for your feedback. I have addressed as indicated. Feel free to take a look again 🙂

**Review by vaurdan — APPROVED** [review]
Sorry for the late review!

Looks great, approved.

**Pull request merged**